### DISCUSSION TITLE
How to unit test a service?

### DISCUSSION THREAD

**Discussion by parijke** [symfony/symfony#42541] [Q&A] [answered]
I tried this https://symfony.com/blog/new-in-symfony-4-1-simpler-service-testing but that didn't work (maybe because I am on 5.3)

So I now test the service instantiating it in the test, but that seems wrong. Any suggestions for improvements?

```
<?php

namespace App\Tests;

use App\Service\BookingsEngineService;
use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;

class BookingsEngineTest extends KernelTestCase
{
    public function testSomething(): void
    {
        self::bootKernel();

        $engine = new BookingsEngineService();
        $result = $engine->getObjectIds();

        $this->assertContains('60294', $result);
    }
}
```

**Comment by parijke** [comment, marked as answer]
Sorry https://symfony.com/doc/current/testing.html#retrieving-services-in-the-test

**Reply by tarlepp** [reply]
imho - making `$engine = new BookingsEngineService();` is no problem, you're making that service from scratch - you don't need container there right? Also that `self::bootKernel();` is obsolete in this use case. With this you separate your actual service from everything else, which is good - because then you don't get any side effects from other parts.
